DifferenceGenerator/src/test/kotlin/TestCaseGenerator.kt

```diff
@@ -0,0 +1,161 @@
+     * sum to 1.
+     *
+     * @property transitionProbabilities The map of transition probabilities.
+     */
+    private val transitionProbabilities: Map<AlignmentElement, Map<AlignmentElement, Float>> =
+        mapOf(
+            AlignmentElement.INSERTION to
+                mapOf(
+                    AlignmentElement.INSERTION to
+                        0.3f,
+                    AlignmentElement.DELETION to
+                        0.05f,
+                    AlignmentElement.MATCH to
+                        0.65f,
+                ),
+            AlignmentElement.DELETION to
+                mapOf(
+                    AlignmentElement.INSERTION to
+                        0.05f,
+                    AlignmentElement.DELETION to
+                        0.2f,
+                    AlignmentElement.MATCH to
+                        0.75f,
+                ),
+            AlignmentElement.MATCH to
+                mapOf(
+                    AlignmentElement.INSERTION to
+                        0.1f,
+                    AlignmentElement.DELETION to
+                        0.05f,
+                    AlignmentElement.MATCH to
+                        0.85f,
+                ),
+        )
+
+    /**
+     * Generates a random test case.
+     * @return The Alignment of the two videos.
+     */
+    fun generateRandomTestCase(): Array<AlignmentElement> {
+        val videoGenerator1 = VideoGeneratorImpl(path1)
+        val videoGenerator2 = VideoGeneratorImpl(path2)
+
+        var position = 0
+        var result = ArrayList<AlignmentElement>()
+
+        var currentElement = AlignmentElement.MATCH
+
+        while (position < baseLength) { // + Random.nextInt(-2, 2)
+            // use transitionProbabilites to determine next element
+            // generate random float between 0 and 1
+            val randomFloat = Random.nextFloat()
+
+            // get the next element based on the random float
+            val nextElement =
+                when {
+                    randomFloat <
+                        transitionProbabilities[
+                            currentElement,
+                        ]!![
+                            AlignmentElement.INSERTION,
+                        ]!! -> {
+                        AlignmentElement.INSERTION
+                    }
+                    randomFloat <
+                        transitionProbabilities[
+                            currentElement,
+                        ]!![
+                            AlignmentElement.INSERTION,
+                        ]!! +
+                        transitionProbabilities[
+                            currentElement,
+                        ]!![
+                            AlignmentElement.DELETION,
+                        ]!! -> {
+                        AlignmentElement.DELETION
+                    }
+                    else -> {
+                        AlignmentElement.MATCH
+                    }
+                }
```

Comment:
want to say a markov-chain and then just step through it, but we will never use it again probably hence this solution is the shortest